### DISCUSSION TITLE
CanMatrix dependency

### DISCUSSION THREAD

**Discussion by e-roux** [danielhrisca/asammdf#652] [General]
Hello,

one of the dependency in the requirements.txt is CanMatrix. It seems to be used exclusively in `get_(bus|can|lin)_signal`, so is this really a requirement or should it be a pkg extra? 

The question is valid for pandas too.

**Comment by danielhrisca** [comment]
Hello Emmanuel,

the bus logging stuff is part of the MDF v4 specification. Any reason why having it installed by default would be a problem?

**Reply by e-roux** [reply]
Hi Daniel,
the MDF spec is big, I personally do not use those logging very often. Putting all the functionalities in a big package drive to a lot of dependencies, and `CanMatrix` bring its own, e.g. `click` or `attrs`. Putting `pandas` in `asammdf` drive finally to an expensive requirement when I sometime only need to read a few channels. As your data blocks are managed with numpy, I guess it coud be the only requirement. `numexpr` as far as I know (do not know a lot this lib) takes advantage of the intel mkl lib but may make sense with conda, with pip not sure.  Parts of the spec could be in `extra_requires` section of setup.cfg without impacting the fact that the spec is implemented.

**Reply by danielhrisca** [reply]
Some of the packages are used now in the function annotations so I think it would be really hard to make them optional at this point. 

``numexpr`` is used for algebraic conversions

**Reply by e-roux** [reply]
ok, thanks for the answer
